spot-client/src/spot-remote/ui/loaders/withRemoteControl.js

```diff
@@ -209,50 +175,18 @@ export class RemoteControlLoader extends AbstractLoader {
                 return Promise.reject();
             }
 
-            return this._reconnect();
+            // This will trigger reconnect
+            throw error;
         });
     }
 
     /**
-     * Attempts to re-establish a previous connection to the remote control
-     * service.
+     * Disconnects the remote control service.
      *
-     * @private
      * @returns {Promise}
      */
-    _reconnect() {
-        // Underneath remoteControlService, strophe can end up disconnecting
-        // twice, and having two reconnects should be avoided.
-        if (this._isReconnectQueued) {
-            logger.warn('Reconnect called while already reconnecting');
-
-            return Promise.reject();
-        }
-
-        if (this._unmounted) {
-            logger.warn('Cancelling reconnect due to unmount');
-
-            return Promise.reject();
-        }
-
-        this._isReconnectQueued = true;
-
-        // wait a little bit to retry to avoid a stampeding herd
```

Comment:
I surprised myself that I account for this. I'm really proud of myself and I'm letting you know that I am.